app/controllers/admin/attachments_controller.rb

```diff
@@ -201,13 +201,11 @@ def handle_duplicate_key_errors_caused_by_double_create_requests(exception)
   def save_attachment
     result = attachment.save(context: :user_input)
 
-    if result && attachment.is_a?(HtmlAttachment)
-      Whitehall::PublishingApi.save_draft(attachment)
-    end
-
     if attachable_is_an_edition?
       draft_updater = Whitehall.edition_services.draft_updater(attachable)
       draft_updater.perform!
+    elsif result && attachment.is_a?(HtmlAttachment)
+      Whitehall::PublishingApi.save_draft(attachment)
```

Comment:
Could this commit be moved to be before all the others?  Then you won't need to add the extra expectations to the test, then remove them again.